src/crypto/blowfish.cr

```diff
@@ -8,8 +8,11 @@ class Crypto::Blowfish
   @s3 : UInt32*
 
   def initialize(@rounds = DEFAULT_ROUNDS)
-    @p = P.dup
-    @s = S.dup
+    @p = uninitialized UInt32[18]
+    @s = uninitialized UInt32[1024]
+
+    @p.to_slice.copy_from(P.to_slice)
+    @s.to_slice.copy_from(S.to_slice)
```

Comment:
Shouldn't we do the same for `CYPHER_TEXT` in `Crypto::Bcrypt#hash_password`?

Reply:
It's just 24 bytes, I don't think that would have any observable impact on build or run times

Reply:
It's still a pointless call to the GC at runtime.

Reply:
I'd consider this a regression.

Previously, `CYPHER_TEXT` as a static array, so `dup` didn't have any effect. It would already be copied to the stack anyway. Now that we change `CYPHER_TEXT` to a slice, `dup` would introduce a new (and unnecessary) heap allocation.

Reply:
We could probably leave this one as a `StaticArray` then. We could even construct the array in the method body itself to remove the constant initializer overhead.

Reply:
Indeed, we don't _need_ the constant.

Reply:
We don't need it. But it's still nice to put explicit names on things. And to extract the data definition out of the algorithm. 

We technically don't need `P` and `S` either. But I don't think we would like to inline them.

The constant initializer overhead should go away with true slice literals.